cmd/mdatagen/main_test.go

```diff
@@ -58,8 +58,10 @@ metrics:
 type: metricreceiver
 status:
   class: exporter
-  stability: beta
-  pipelines: [metrics, logs, traces]
+  stability:
+    metrics: beta
+    logs: beta
+    traces: beta
```

Comment:
Do we want to require specifying the stability per signal all the time?

What if we have 

```suggestion
  stability:
    overall: beta
    per_signal:
      metrics: beta
      logs: beta
      traces: beta
```

Making `stability.overal` the only required field, which can be overridden by `per_signal` for particular signals? This can still generate the `<Type>Stability` constants.

WDYT?

Reply:
`overall` maybe not the best word, just a first one came to my mind

Reply:
The nice thing about having to specify it per signal is that it allows us to remove the need for a "supported pipelines" field. I agree that it's a bit repetitive though in the case where all signals have the same stability. 🤷🏻

Reply:
Potentially repetitive, but ultimately exactly accurate.

Reply:
> The nice thing about having to specify it per signal is that it allows us to remove the need for a "supported pipelines" field. I agree that it's a bit repetitive though in the case where all signals have the same stability. 🤷🏻

This yaml also can be rendered as it's done in this PR without "supported pipelines". Not sure if it's the best for the end user though, given extensions have to be rendered differently... Maybe representation closer to the suggested metadata.yaml is better, something like:
```
| Status                   |                              |
| ------------------------ | ---------------------------- |
| Stability                | [alpha]                      |
| Supported pipeline types | logs, metrics [beta], traces |
| Distributions            | [contrib]                    |
```

cc @open-telemetry/collector-contrib-approvers please share your thoughts

Reply:
I agree that specifying stability per-signal feels repetitive, but I think it's probably the best solution. A quick search in contrib shows that ~100 components only support one signal type, ~30 support all three, and ~10 support two. So for most existing components an `overall` key won't save on repetition too much. Of course, as more signals are added in the future, it could help with components that support a lot of signals.

One idea to simplify things just a little from a configuration perspective could be to key on stability instead of on signal type. For example:
```yaml
stability:
  beta: [metrics, logs]
  alpha: [traces]
```

Reply:
>This yaml also can be rendered as it's done in this PR without "supported pipelines". Not sure if it's the best for the end user though, given extensions have to be rendered differently... Maybe representation closer to the suggested metadata.yaml is better, something like

@dmitryax I'm worried that there would be some ambiguity in how to read this table. An alternate way I could read the example you gave is that the component itself has alpha stability, its logs and metrics capabilities are beta, and traces is a "default" stability (stable?). I think we should probably default to being more explicit even if it is a bit noisier.

Reply:
```
stability:
  alpha: [traces]
  beta: [metrics, logs]
```

I really like the way this looks, would be great if we could view it like that in the README as well.

Reply:
```yaml
stability:
  alpha: [traces]
  beta: [metrics, logs]
```

Looks good to me as well. @codeboten WDYT?

Reply:
This format works for me, I'll update this PR accordingly 👍🏻

Reply:
I've updated the PR to follow this format, @TylerHelmuth @dmitryax @evan-bradley PTAL. If everyone is ok with this, i'll update the PR to add the metadata files that have been added to main since this PR has been opened.